src/main/java/utils/HttpRequestPath.java

```diff
@@ -0,0 +1,57 @@
+package utils;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class HttpRequestPath {
+    private static final String PATH_PARSER_TOKEN_WITH_ESCAPE = "\\?";
+    private static final String PATH_PARSER_TOKEN = "?";
+    private static final String QUERY_STRING_PARSER_TOKEN = "&";
+    private static final String KEY_VALUE_PARSER_TOKEN = "=";
+
+    private final String path;
+    private final HttpRequestQueryString httpRequestQueryString;
+
+    public HttpRequestPath(String path, HttpRequestQueryString httpRequestQueryString) {
+        this.path = path;
+        this.httpRequestQueryString = httpRequestQueryString;
+    }
+
+    public static HttpRequestPath findHttpRequestPath(String pathString) {
+        Map<String, String> httpRequestQueryStrings
+            = new HashMap<>();
```

Comment:
줄바꿈을 할 필요가 없을 것 같아요. 🙂
```suggestion
        Map<String, String> httpRequestQueryStrings = new HashMap<>();
```